Add Vite (client only, no SSR) build. Update Next.js 14.2.2 and Remix to 2.8.0 (#5970)
plone/volto: docs/source/contributing/developing-core.md

## Supported frameworks

> Plone supports several frontend implementations, the main one being Volto as the default frontend and reference React-based implementation.

Reviewer: ```suggestion
Plone supports several frontend implementations.
Volto, a React-based implementation, is the default frontend for Plone 6.
```

Author: @stevepiercy I'd like to start dropping the "reference implementation" concept here and there. We talked about it during the sprint, if you think that the wording is not correct and we should find a better one, let's do it, but I'd like to convey this from now on for Volto and Classic.

Reviewer: Oh, I thought it was a mistake, not intentional. In that case the term is "reference implementation" and should not have other terms inserted within. I updated my suggestion accordingly.

I added the term to the Documentation glossary in https://github.com/plone/documentation/pull/1656.

> The others are currently under heavy development, marked as experimental and for now, they remain as a proof of concept.

Reviewer: ```suggestion

Other frontends are currently under heavy development.
They are marked as experimental and, for now, they are a proof of concept demonstrating that other frontends are possible.
```

Although they do work now in an acceptable way, the implementation might change in the future.
Please note that these implementations only show how to access the public Plone content in the current site and dealing with data fetching and routing.

> They use the Plone Frontend Strategic Packages (`@plone/registry`, `@plone/client`, `@plone/components`, etc).

Reviewer: ```suggestion
They use the Plone frontend strategic packages, including `@plone/registry`, `@plone/client`, and `@plone/components`.
```

### Plone

> The default frontend and reference React-based implementation in Plone is Volto.

Reviewer: ```suggestion
Volto is the default frontend for Plone 6.
It is a React-based implementation.
```

Author: Same as above

Reviewer: Suggestion updated.

> In the `apps` folder you'll find a Volto project scaffolding that uses Volto as a library.

Reviewer: ```suggestion
In the `apps` folder, you'll find a Volto project scaffolding that uses Volto as a library.
```

Reviewer: @sneridagh can you verify whether this sentence and the next one are still correct, especially since we're kicking Yeoman to the curb.

Author: It is correct for now, we will have to change it (or even remove it) when the new setup is a official.

This is the same as the one that you'll have when running the Volto generator or `cookiecutter-plone-starter`.

### Next.js

> This is the proof of concept using Next.js with Plone.

Reviewer: ```suggestion
This frontend is a proof of concept using Next.js with Plone.
```

> You can try it out using:

Reviewer: ```suggestion
You can try it out using the following command.
```

> ```

Reviewer: ````suggestion
```shell
````

pnpm --filter plone-nextjs dev
```

### Remix

> This is the proof of concept using Remix with Plone.

Reviewer: ```suggestion
This frontend is a proof of concept using Remix with Plone.
```

> You can try it out using:

Reviewer: ```suggestion
You can try it out using the following command.
```

> ```

Reviewer: ````suggestion
```shell
````

pnpm --filter plone-remix dev
```

### Vite build (client only)

> This is the proof of concept using a custom client build based in Vite with Plone.

Reviewer: ```suggestion
This frontend is a proof of concept using a custom client build based in Vite with Plone.
```

> It uses `@tanstack/router` in combination with `@plone/client` (which in turns uses `@tanstack/query`).

Reviewer: ```suggestion
It uses `@tanstack/router` in combination with `@plone/client`, which in turns uses `@tanstack/query`.
```

This build is suitable for applications that do not need server side generation, and it's client only.

> You can try it out using:

Reviewer: ```suggestion
You can try it out using the following command.
```

> ```

Reviewer: ````suggestion
```shell
````

pnpm --filter plone-vite dev
```

### Vite SSR build

> This is the proof of concept using a custom build based in Vite with SSR with Plone.

Reviewer: ```suggestion
This frontend is a proof of concept using a custom build, based in Vite with SSR with Plone.
```

It uses `@tanstack/router` in combination with `@plone/client` (which in turns uses `@tanstack/query`).

> You can try it out using:

Reviewer: ```suggestion
You can try it out using the following command.
```

> ```

Reviewer: ````suggestion
```shell
````
